packages/toolpad-app/src/toolpad/AppEditor/PageEditor/RenderPanel/RenderOverlay.tsx

```diff
@@ -203,51 +203,51 @@ function deleteOrphanedLayoutNodes(
           moveTargetNodeId !== parentParent.id &&
           moveTargetNodeId !== lastContainerChild.id
         ) {
-          draftDom = appDom.moveNode(
-            draftDom,
-            lastContainerChild,
-            parentParentParent,
-            lastContainerChild.parentProp,
-            parentParent.parentIndex,
-          );
-
-          if (isPageColumn(parentParent)) {
-            draftDom = appDom.setNodeNamespacedProp(
+          if (
+            moveTargetNodeId !== parent.id &&
+            moveTargetNodeId !== lastContainerChild.id &&
+            isPageLayoutComponent(parentParent)
+          ) {
+            draftDom = appDom.moveNode(
               draftDom,
               lastContainerChild,
-              'layout',
-              'columnSize',
-              parentParent.layout?.columnSize || appDom.createConst(1),
+              parentParent,
+              lastContainerChild.parentProp,
+              parent.parentIndex,
             );
-          }
 
-          orphanedLayoutNodeIds = [...orphanedLayoutNodeIds, parentParent.id];
-        }
+            if (isPageColumn(parent)) {
+              draftDom = appDom.setNodeNamespacedProp(
+                draftDom,
+                lastContainerChild,
+                'layout',
+                'columnSize',
+                parent.layout?.columnSize || appDom.createConst(1),
+              );
+            }
+
+            orphanedLayoutNodeIds = [...orphanedLayoutNodeIds, parent.id];
+          }
 
-        if (
```

Comment:
I just moved this `if` block to above, the diff makes it seems like more than that for some reason.